lib/puppet/resource.rb

```diff
@@ -88,6 +88,9 @@ def to_data_hash(json_serializer = nil)
         value = Puppet::Resource.value_to_pson_data(value)
         if is_json_type?(value)
           params[param] = value
+        elsif json_serializer.nil?
+          Puppet.warning("Resource '#{to_s}' contains a #{value.class.name} value. It will be converted to the String '#{value}'")
```

Comment:
I tried. Symbols are accepted, but they are converted to strings. They do not retain the type. This is what you get with PSON:
```
2.1.5 :001 > require 'puppet'
 => true 
2.1.5 :002 > PSON.parse({ :x => 'hello' }.to_pson)
 => {"x"=>"hello"}
2.1.5 :003 > PSON.parse({ :x => :undef }.to_pson)
 => {"x"=>"undef"} 
```